cartography/intel/gsuite/api.py

```diff
@@ -0,0 +1,243 @@
+import logging
+
+from googleapiclient.discovery import HttpError
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_all_groups(admin):
+    """
+    Return list of Google Groups in your organization
+    Returns empty list if we are unable to enumerate the groups for any reasons
+
+    :param admin: apiclient discovery resource object
+    See
+    :return: List of Google groups in domain
+
+    """
+    request = admin.groups().list(customer='my_customer', maxResults=20, orderBy='email')
+    groups = []
+    while request is not None:
```

Comment:
Believe it's needed because it's a paginated API.  `request` is `None` if there are no results to be returned, which is used by https://github.com/lyft/cartography/blob/d636154174794d9851924c4a8664f62403539cbc/cartography/intel/gsuite/api.py#L29